config/routes.rb

```diff
@@ -26,7 +26,12 @@
     get "/locale/:locale", to: "stories#index"
 
     draw :admin
-    draw :listing
+
+    # The lambda (e.g. `->`) allows for dynamic checking.  In other words we check with each
+    # request.
+    constraints(->(_req) { Listing.feature_enabled? }) do
```

Comment:
I'm still a bit unsure on why we added `.feature_enabled?` to `Listing`, because I think the class itself doesn't need to know about feature flags. We didn't add `Listing.enable_feature` and `Listing.disable_feature` as wrappers around `FeatureFlag`, why did we do it for this method? I feel this constraint would be just as clear:

```suggestion
    constraints(->(_req) { FeatureFlag.accessible?(:listing_feature_enabled) }) do
```

On a related note: it's a bit odd to have the `_enabled` suffix on the flag itself, I don't think we did this for other feature flags.

Anyway, none of this is blocking in anyway, just my random ramblings 😂

Reply:
The one bit of "why the indirection" is that the default behavior during this "silent rollout" was to assume the listings were enabled for all sites.  But then to flip that assumption once we've validated.  The indirection of `Listing.feature_enabled?` means I can toggle that "assumption" (e.g. from `FeatureFlag.accessible?(:listing_feature_enabled)` to the version that assumes it's not accessible unless explicitly stated.  

The last bit on the "method instead of symbol parameter" is that it's easy to introduce types on the symbol passed (or be inconsistent in naming).  Whereas the method allows us to rely on the parser just a bit more.

I do agree that the `listing_feature_enabled` is poorly labeled.

Reply:
Got it, thanks for explaining.

> the default behavior during this "silent rollout" was to assume the listings were enabled for all sites. [...] I can toggle that "assumption" (e.g. from FeatureFlag.accessible?(:listing_feature_enabled) to the version that assumes it's not accessible unless explicitly stated.

Hmm. I feel like we could have achieved the same by running `FeatureFlag.enable` on the original deployment which will automatically create the flag if it doesn't exist yet. Toggling could then have been done with either `FeatureFlag.disable` or `FeatureFlag.remove`. I don't think that there's much difference between the two approaches, both need deployments. Anyway, mostly a pointless discussion, sometimes my brain gets stuck on these things, my apologies.

> it's easy to introduce types on the symbol passed 

I assume you mean "typos on the symbol parameter", not "types", right? Tbh, I've been thinking about this just recently. It'd probably be a good idea if we introduced a file where we keep all feature flags names as constants, e.g. `LISTING_FLAG = "listings"` and then only use the constants. This would also give us a single source of truth regarding all currently active flags.